app/presenters/hyrax/work_show_presenter.rb

```diff
@@ -132,8 +132,13 @@ def export_as_ttl
       graph.dump(:ttl)
     end
 
+    ##
+    # @deprecated use `::Ability.can?(:edit, presenter)`. Hyrax views calling
+    #   presenter {#editor} methods will continue to call them until Hyrax
+    #   4.0.0. The deprecation time horizon for the presenter methods themselves
+    #   is 5.0.0.
     def editor?
-      current_ability.can?(:edit, solr_document)
+      current_ability.can?(:edit, self)
```

Comment:
Do you want to add a deprecation warning?  It might make it easier to refactor the existing application.

Reply:
i think the deprecation warning would just be annoying. Hyrax will need to continue to call these methods until 4.0.0, so we'd get a lot of chatter in the logs from local calls.